src/application/services/crux-asset-translator.ts

```diff
@@ -0,0 +1,103 @@
+import { IAddress, IAddressMapping } from "../../core/entities/crux-user";
+import { errors } from "../../packages";
+import { IGlobalAssetList } from "../../packages/configuration-service";
+import { getLogger } from "../../packages/logger";
+const log = getLogger(__filename);
+
+export interface IClientAssetMapping {
+    [currencySymbol: string]: string;
+}
+
+export interface IGlobalAsset {
+    assetId: string;
+    symbol: string;
+    name: string;
+    assetType: string|null;
+    decimals: number|null;
+    assetIdentifierName: string|null;
+    assetIdentifierValue: number|string|null;
+    parentAssetId: string|null;
+}
+
+export interface IResolvedClientAssetMap {
+    [currencySymbol: string]: IGlobalAsset;
+}
+
+export interface IReverseClientAssetMapping {
+    [assetId: string]: string;
+}
+
+export interface IPutAddressMapSuccess {
+    [currency: string]: IAddress;
+}
+
+export interface IPutAddressMapFailures {
+    [currency: string]: string;
+}
+
+export class CruxAssetTranslator {
+    private _assetMapping: IClientAssetMapping;
+    private _reverseAssetMap: IReverseClientAssetMapping;
+    constructor(assetMapping: IClientAssetMapping) {
+        this._assetMapping = assetMapping;
```

Comment:
this would be the right place to convert the keys to lowercase and do validation of the assetMapping given as input.

Reply:
Fixed and updated

Reply:
assetMapping needs validation before initialising the class successfully